src/lib/pages/workflow-history-json.svelte

```diff
@@ -3,21 +3,45 @@
 
   import CodeBlock from '$lib/holocene/code-block.svelte';
   import Loading from '$lib/holocene/loading.svelte';
-  import { fetchRawEvents } from '$lib/services/events-service';
+  import ToggleSwitch from '$lib/holocene/toggle-switch.svelte';
+  import { fetchAllEvents, fetchRawEvents } from '$lib/services/events-service';
+  import { authUser } from '$lib/stores/auth-user';
   import { decodeURIForSvelte } from '$lib/utilities/encode-uri';
 
   const { namespace, workflow: workflowId, run: runId } = $page.params;
 
-  const events = fetchRawEvents({
-    namespace: decodeURIForSvelte(namespace),
-    workflowId: decodeURIForSvelte(workflowId),
-    runId: decodeURIForSvelte(runId),
-    sort: 'ascending',
-  });
+  let decodeEventHistory = true;
+
+  $: events = decodeEventHistory
+    ? fetchAllEvents({
+        namespace: decodeURIForSvelte(namespace),
+        workflowId: decodeURIForSvelte(workflowId),
+        runId: decodeURIForSvelte(runId),
+        settings: $page.data.settings,
+        accessToken: $authUser?.accessToken,
+        sort: 'ascending',
+      })
+    : fetchRawEvents({
+        namespace: decodeURIForSvelte(namespace),
+        workflowId: decodeURIForSvelte(workflowId),
+        runId: decodeURIForSvelte(runId),
+        sort: 'ascending',
+      });
 </script>
 
 {#await events}
   <Loading />
 {:then events}
+  <div class="flex justify-end">
+    <label
+      for="decode-event-history"
+      class="flex items-center gap-4 font-secondary text-sm"
+      >View Decoded Event History<ToggleSwitch
```

Comment:
Not sure if this is what we want here, but noticed elsewhere (e.g. `Auto refresh`) we lowercase following words. 

```suggestion
      >View decoded event history<ToggleSwitch
```